### PR TITLE
feat(react): allow pagination labels to take react nodes

### PR DESCRIPTION
**Pull request by thuey**
Closes https://github.com/dequelabs/cauldron/issues/955

### CODE REVIEW

File: packages/react/src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -9,10 +9,10 @@ interface Props extends React.HTMLAttributes<HTMLDivElement> {
   itemsPerPage?: number;
   currentPage?: number;
   statusLabel?: React.ReactNode;
-  firstPageLabel?: string;
-  previousPageLabel?: string;
-  nextPageLabel?: string;
-  lastPageLabel?: string;
+  firstPageLabel?: React.ReactNode;
```
Comment (scurker) [inline packages/react/src/components/Pagination/Pagination.tsx:12]: I think `React.ReactNode` is more inclusive of what we actually want:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/59c89cc048de1e795eb3c88108fcb982ee4674ae/types/react/v16/index.d.ts#L235

I don't think we want people to be able to pass in `boolean`, `null`, or `undefined`. What do you think?

Comment (scurker) [reply packages/react/src/components/Pagination/Pagination.tsx:12]: Hmm... that would be a breaking change for that component. I think `Omit<ReactNode, boolean | null | undefined>` is probably what we want here (and elsewhere). If we're going to use a more loose type, we should ensure that it's helping us catch edge case issues. Do you mind creating a ticket for `IconButton` and see if we can implement the constrained type here?

Comment (thuey) [reply packages/react/src/components/Pagination/Pagination.tsx:12]: Sure. Do you want a separate ticket for [statusLabel](https://github.com/dequelabs/cauldron/blob/develop/packages/react/src/components/Pagination/Pagination.tsx#L11), too?

**Review by scurker — APPROVED** [review]

File: packages/react/src/components/Pagination/Pagination.tsx (hunk 1 above)
Comment (thuey) [reply packages/react/src/components/Pagination/Pagination.tsx]: https://github.com/dequelabs/cauldron/issues/958
https://github.com/dequelabs/cauldron/issues/957

**Pull request merged**